### PR TITLE
create new module for table

### PR DESCRIPTION
**Pull request by Nuwa-Hub**
[ITP-1568]

[ITP-1568]: https://fcodelabs.atlassian.net/browse/ITP-1568?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by Nuwa-Hub** [comment]
Screenshot 2023-02-03 095442

File: src/app/table-module/services/table.service.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    }
+
+    public save(data: Product[], isNew?: boolean): void {
+        const action = isNew ? CREATE_ACTION : UPDATE_ACTION;
+
+        this.reset();
+
+        this.fetch(action, data)
+            .subscribe(() => this.read(), () => this.read());
+    }
```
Comment (ChamaraWee) [inline src/app/table-module/services/table.service.ts]: purpose of two `this.read()` ?

File: src/app/table-module/services/table.service.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+    public remove(data: Product[]): void {
+        this.reset();
+
+        this.fetch(REMOVE_ACTION, data)
+            .subscribe(() => this.read(), () => this.read());
```
Comment (ChamaraWee) [inline src/app/table-module/services/table.service.ts]: purpose of two this.read() ?

File: src/app/table-module/table-module.module.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+import { ReactiveFormsModule } from '@angular/forms';
+
+import { GridModule } from '@progress/kendo-angular-grid';
+
+import { AppComponent } from './app.component';
+import { EditService } from './edit.service';
+
+
+
+const config: SocketIoConfig = { url: "http://localhost:5000", options: {} };
```
Comment (ChamaraWee) [inline src/app/table-module/table-module.module.ts]: move urls to env

File: src/app/table-module/table-module.module.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        HttpClientModule,
+        HttpClientJsonpModule,
+        BrowserModule,
+        BrowserAnimationsModule,
+        ReactiveFormsModule,
+        GridModule
+    ],
+    providers: [
+        {
+            deps: [HttpClient],
```
Comment (ChamaraWee) [inline src/app/table-module/table-module.module.ts]: not correct implementation ask from Thilina how to implement this.

File: src/app/table-module/containers/home-page/home-page.component.ts [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+        sort: [],
+        skip: 0,
+        take: 5
+    };
+    public formGroup: FormGroup;
+
+    private editService: EditService;
+    private editedRowIndex: number;
+
+    constructor(@Inject(EditService) editServiceFactory: () => EditService) {
```
Comment (ChamaraWee) [inline src/app/table-module/containers/home-page/home-page.component.ts]: way you use to inject dependency is not correct.

**Review by ChamaraWee — CHANGES_REQUESTED** [review]

File: src/app/table-module/services/table.service.ts (hunk 1 above)
Comment (Nuwa-Hub) [reply src/app/table-module/services/table.service.ts]: im not doin any thin this is the fuction that in the kendo angular file

File: src/app/table-module/services/table.service.ts (hunk 2 above)
Comment (Nuwa-Hub) [reply src/app/table-module/services/table.service.ts]: im not doin any thin this is the fuction that in the kendo angular file

File: src/app/table-module/containers/home-page/home-page.component.ts (hunk 5 above)
Comment (Nuwa-Hub) [reply src/app/table-module/containers/home-page/home-page.component.ts]: remove the factory method

File: src/app/table-module/table-module.module.ts (hunk 4 above)
Comment (Nuwa-Hub) [reply src/app/table-module/table-module.module.ts]: corect

File: src/app/table-module/services/table.service.ts (hunk 2 above)
Comment (Nuwa-Hub) [reply src/app/table-module/services/table.service.ts]: remove  that

File: src/app/table-module/services/table.service.ts (hunk 1 above)
Comment (Nuwa-Hub) [reply src/app/table-module/services/table.service.ts]: remove it

File: src/app/table-module/table-module.module.ts (hunk 3 above)
Comment (Nuwa-Hub) [reply src/app/table-module/table-module.module.ts]: done

Comment (ChamaraWee) [reply src/app/table-module/table-module.module.ts]: once you move this to env. You have to get the URL from env.

**Review by ChamaraWee — CHANGES_REQUESTED** [review]

**Review by ChamaraWee — APPROVED** [review]

**Pull request merged**